### PR TITLE
Refactoring of generator.js

### PR DESCRIPTION
**Pull request by aayusharora**

### CODE REVIEW

File: src/generator/backend/fold.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+function speakerNameWithOrg(speaker) {
+  return speaker.organisation ?
+    `${speaker.name} (${speaker.organisation})` :
+    speaker.name;
+}
+
+
+function byProperty(key) {
+  return (a, b) => {
```
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Identifier name 'a' is too short. (< 3)](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l21)

File: src/generator/backend/fold.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+
+function speakerNameWithOrg(speaker) {
+  return speaker.organisation ?
+    `${speaker.name} (${speaker.organisation})` :
+    speaker.name;
+}
+
+
+function byProperty(key) {
```
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Missing JSDoc comment.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l20)

File: src/generator/backend/fold.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    }
+
+    return 0;
+  };
+}
+
+
+function slugify(str) {
+  if (typeof str === 'undefined') {
+    return '';
```
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l36)

File: src/generator/backend/fold.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+  }
+  return str.replace(/[^\w]/g, '-').replace(/-+/g, '-').toLowerCase();
+}
+
+function zeroFill(num) {
+  if (typeof num === 'undefined') {
```
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l42)

File: src/generator/backend/fold.js [hunk 5]
```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+
+var exports = module.exports = {};
+
+const fs = require('fs-extra');
+const moment = require('moment');
+const handlebars = require('handlebars');
```
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l8)

File: src/generator/backend/fold.js [hunk 6]
```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+
+var exports = module.exports = {};
+
+const fs = require('fs-extra');
```
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: ["fs" is defined but never used](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l6)

File: src/generator/backend/fold.js [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+function zeroFill(num) {
+  if (typeof num === 'undefined') {
+    return '';
+  }
+  if (num >= 10) {
+    return num.toString();
+  }
+  return '0' + num.toString();
+}
+function foldByTrack(sessions, speakers) {
```
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l50)

File: src/generator/backend/fold.js [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+
+
+function slugify(str) {
+  if (typeof str === 'undefined') {
+    return '';
+  }
+  return str.replace(/[^\w]/g, '-').replace(/-+/g, '-').toLowerCase();
+}
+
+function zeroFill(num) {
```
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Use the function form of "use strict".](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l41)

Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Missing JSDoc comment.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l41)

File: src/generator/backend/fold.js [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+    return 0;
+  };
+}
+
+
+function slugify(str) {
+  if (typeof str === 'undefined') {
+    return '';
+  }
+  return str.replace(/[^\w]/g, '-').replace(/-+/g, '-').toLowerCase();
```
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737479688/issues/source?bid=3233857&fileBranchId=3352811#l38)

File: src/generator/backend/generator.js [hunk 10]
```diff
@@ -7,6 +7,7 @@ const moment = require('moment');
 const handlebars = require('handlebars');
 const async = require('async');
 const archiver = require('archiver');
+const fold=require('./fold.js');
```
Comment (championswimmer) [inline src/generator/backend/generator.js]: spaces around `=`

Comment (championswimmer) [inline src/generator/backend/generator.js]: also prefer using (__dirname + '/fold.js') as we never know where the generator.js is called from. __dirname ensures dirpath starts from the folder containing this file

File: src/generator/backend/fold.js (hunk 2 above)
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Missing JSDoc comment.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l20)

File: src/generator/backend/fold.js (hunk 1 above)
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Identifier name 'a' is too short. (< 3)](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l21)

File: src/generator/backend/fold.js (hunk 3 above)
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l36)

File: src/generator/backend/fold.js (hunk 4 above)
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l42)

File: src/generator/backend/fold.js (hunk 5 above)
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l8)

File: src/generator/backend/fold.js (hunk 6 above)
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: ["fs" is defined but never used](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l6)

File: src/generator/backend/fold.js (hunk 7 above)
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l50)

File: src/generator/backend/fold.js (hunk 8 above)
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Use the function form of "use strict".](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l41)

Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Missing JSDoc comment.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l41)

File: src/generator/backend/fold.js (hunk 9 above)
Comment (juslee) [inline src/generator/backend/fold.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2737543607/issues/source?bid=3233857&fileBranchId=3352811#l38)

**Comment by championswimmer** [comment]
okay looks good to me.

**Pull request merged**